pymatgen/command_line/mcsqs_caller.py

```diff
@@ -29,93 +34,222 @@ def run_mcsqs(
     scaling: Union[int, List[int]],
-    search_time: float = 0.01,
-) -> Optional[Sqs]:
+    search_time: float = 60,
+    directory: Optional[str] = None,
+    instances: Optional[int] = None,
+    temperature: Union[int, float] = 1,
+    wr: float = 1,
+    wn: float = 1,
+    wd: float = 0.5,
+    tol: float = 1e-3
+) -> Sqs:
     """
     Helper function for calling mcsqs with different arguments
     Args:
-        structure (Structure): disordered pymatgen Structure object
-        clusters (dict): dictionary of cluster interactions with entries in the form
+        structure (Structure): Disordered pymatgen Structure object
+        clusters (dict): Dictionary of cluster interactions with entries in the form
             number of atoms: cutoff in angstroms
-        scaling (int or list): scaling factor to determine supercell. Two options are possible:
+        scaling (int or list): Scaling factor to determine supercell. Two options are possible:
                 a. (preferred) Scales number of atoms, e.g., for a structure with 8 atoms,
                    scaling=4 would lead to a 32 atom supercell
-                b. An sequence of three scaling factors, e.g., [2, 1, 1], which
+                b. A sequence of three scaling factors, e.g., [2, 1, 1], which
                    specifies that the supercell should have dimensions 2a x b x c
-        search_time (int): The time spent looking for the ideal SQS in minutes
+    Keyword Args:
+        search_time (float): Time spent looking for the ideal SQS in minutes (default: 60)
+        directory (str): Directory to run mcsqs calculation and store files (default: None
+            runs calculations in a temp directory)
+        instances (int): Specifies the number of parallel instances of mcsqs to run
+            (default: number of cpu cores detected by Python)
+        temperature (int or float): Monte Carlo temperature (default: 1), "T" in atat code
+        wr (int or float): Weight assigned to range of perfect correlation match in objective
+            function (default = 1)
+        wn (int or float): Multiplicative decrease in weight per additional point in cluster (default: 1)
+        wd (int or float): Exponent of decay in weight as function of cluster diameter (default: 0.5)
+        tol (int or float): Tolerance for matching correlations (default: 1e-3)
 
     Returns:
-        Tuple of Pymatgen structure, which is an SQS of the input structure, and the
-            mcsqs objective function
+        Tuple of Pymatgen structure SQS of the input structure, the mcsqs objective function,
+            list of all SQS structures, and the directory where calculations are run
     """
 
     num_atoms = len(structure)
 
     if structure.is_ordered:
         raise ValueError("Pick a disordered structure")
 
-    with ScratchDir("."):
+    if instances is None:
+        instances = os.cpu_count()
+
+    # TODO: figure out how to handle instances=1 (doesn't generate bestsqs)
```

Comment:
Is this TODO important?

Reply:
maybe? running with instances=1 should either reset instances to none, or the output of bestsqs1 should be fed directly to create a bestsqs file. in practice instances should never be set to 1, but it may be useful for tests